source/PlayerInfoPanel.cpp

```diff
@@ -559,35 +641,62 @@ void PlayerInfoPanel::DrawFleet(const Rectangle &bounds)
 	Color bright = *GameData::Colors().Get("bright");
 	Color elsewhere = *GameData::Colors().Get("dim");
 	Color dead(.4f, 0.f, 0.f, 0.f);
+	Color special(0.5f, 0.5f, 0.1f, 0.f);
+
+	// Table columns and their starting x positions, alignment and sort comparator
+	columns = {
+		SortableColumn("ship", 0, Table::LEFT, CompareName),
+		SortableColumn("model", 220, Table::LEFT, CompareModelName),
+		SortableColumn("system", 350, Table::LEFT, CompareSystem),
+		SortableColumn("shields", 550, Table::RIGHT, CompareShields),
+		SortableColumn("hull", 610, Table::RIGHT, CompareHull),
+		SortableColumn("fuel", 670, Table::RIGHT, CompareFuel),
+		SortableColumn("crew", 730, Table::RIGHT, CompareRequiredCrew)
+	};
 	
+
 	// Table attributes.
 	Table table;
-	table.AddColumn(0, Table::LEFT);
-	table.AddColumn(220, Table::LEFT);
-	table.AddColumn(350, Table::LEFT);
-	table.AddColumn(550, Table::RIGHT);
-	table.AddColumn(610, Table::RIGHT);
-	table.AddColumn(670, Table::RIGHT);
-	table.AddColumn(730, Table::RIGHT);
+	for(const auto &col: columns)
+		table.AddColumn(col.offset, col.align);
+
 	table.SetUnderline(0, 730);
 	table.DrawAt(bounds.TopLeft() + Point(10., 8.));
-	
+	table.DrawUnderline(hoverMenuPtr == nullptr ? dim : bright);
+
 	// Header row.
-	table.DrawUnderline(dim);
-	table.SetColor(bright);
-	table.Draw("ship");
-	table.Draw("model");
-	table.Draw("system");
-	table.Draw("shields");
-	table.Draw("hull");
-	table.Draw("fuel");
-	table.Draw("crew");
+	
+	for(auto column = columns.begin(); column < columns.end(); ++column)
+	{
+		const auto currentColumn = *column;
+		const Color columnHeaderColor = (hoverMenuPtr == currentColumn.shipSort
+			|| currentSort == currentColumn.shipSort)
+				? bright : dim;
+		const Point tablePoint = table.GetPoint();
+
+		table.Draw(currentColumn.name, columnHeaderColor);
```

Comment:
Probably we are due for an update to the `Table` class, but I admit doing so is outside the scope here. It could then be made responsible for e.g. returning size information about what was just drawn.